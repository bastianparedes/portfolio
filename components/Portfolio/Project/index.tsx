import React from 'react'
import { VscSignOut, VscGithub } from 'react-icons/vsc'
import Image from 'next/image'
import styles from './styles.module.scss'
import stylesRight from './stylesRight.module.scss'
import stylesLeft from './stylesLeft.module.scss'
import classNames from '../../../utils/classNames'

interface ProjectProps {
  index: number
  name: string
  link: string
  image: string
  github: string
  description: string
  technologies: string[]
}

const Project = ({
  index,
  name,
  link,
  image,
  github,
  description,
  technologies
}: ProjectProps): JSX.Element => {
  const stylesSide = index % 2 === 0 ? stylesRight : stylesLeft
  const imagePath = '/images/projects/' + image

  return (
    <div className={styles.projectCard}>
      {link !== '' ? (
        <a
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
          href={link}
          target="_blank"
          rel="noreferrer noopener"
        >
          <Image
            src={imagePath}
            alt={image}
            layout="fill"
            objectFit="contain"
          />
        </a>
      ) : (
        <div
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
        >
          <Image
            src={imagePath}
            alt={image}
            layout="fill"
            objectFit="contain"
          />
        </div>
      )}

      <h2 className={classNames(styles.projectName, stylesSide.projectName)}>
        {name}
      </h2>
      <div
        className={classNames(
          styles.projectDescriptionContainer,
          stylesSide.projectDescriptionContainer
        )}
      >
        <p className={styles.projectDescription}>{description}</p>
      </div>
      <span
        className={classNames(
          styles.projectTechnologies,
          stylesSide.projectTechnologies
        )}
      >
        {technologies.join(', ')}
      </span>
      <div
        className={classNames(styles.linksContainer, stylesSide.linksContainer)}
      >
        {link !== '' && (
          <a
            className={styles.linkWebsite}
            href={link}
            target="_blank"
            rel="noreferrer noopener"
          >
            <VscSignOut />
          </a>
        )}
        {github !== '' && (
          <a
            className={styles.linkGithub}
            href={github}
            target="_blank"
            rel="noreferrer noopener"
          >
            <VscGithub />
          </a>
        )}
      </div>
    </div>
  )
}

export default Project
