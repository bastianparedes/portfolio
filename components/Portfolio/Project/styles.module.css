.projectCard {
  width: 100%;
  max-width: 1000px;
  height: 350px;
  margin: auto;
  display: grid;
  grid-template-columns: repeat(12, 1fr);
  grid-template-rows: repeat(6, 1fr);
  gap: 0;
  justify-items: center;
  align-items: center;
  color: rgb(199, 199, 199);
  position: relative;
}

.imageContainer {
  width: 100%;
  height: 100%;
  grid-row: 1 / -1;
  object-fit: contain;
  background-color: black;
  filter: sepia(1);
  transition: all 0.5s ease;
  position: relative;
}

.imageContainer:hover {
  filter: sepia(0);
  z-index: 2;
}

.projectName {
  width: 100%;
  height: 100%;
  grid-row: 1;
  font-size: 2rem;
  margin: 0;
  color: rgb(255, 255, 255);
}

.projectDescriptionContainer {
  width: 100%;
  height: fit-content;
  grid-row: 2 / 5;
  display: flex;
  align-items: flex-start;
  z-index: 1;
}

.projectDescription {
  background-color: rgb(21, 29, 133);
  color: rgb(199, 199, 199);
  font-size: 1.2rem;
  text-align: justify;
  padding: 1.5rem;
  border-radius: 8px;
  margin: 0;
}

.projectTechnologies {
  width: 100%;
  grid-row: 5;
  font-size: 1rem;
  word-spacing: 0.7rem;
}

.linksContainer {
  width: 100%;
  height: 100%;
  grid-row: 6;
  display: grid;
  align-content: center;
  align-items: center;
  grid-template-columns: repeat(auto-fit, minmax(0, 1fr));
}

.linkGithub,
.linkWebsite {
  width: fit-content;
  font-size: 2rem;
  display: flex;
  text-decoration: none;
  color: rgb(255, 255, 255);
}

.linkGithub:hover,
.linkWebsite:hover {
  transform: scale(1.1);
  color: rgb(0, 255, 255);
}

@media screen and (max-width: 768px) {
  .section {
    padding: 6.25rem 1rem 0 1rem;
  }
  .projectDescription {
    font-size: 1rem;
    padding: 1.3rem;
  }
  .projectName {
    font-size: 1.5rem;
  }
}

@media screen and (max-width: 480px) {
  .section {
    padding: 6.25rem 1rem 0 1rem;
  }
  .projectDescription {
    font-size: 0.9rem;
    padding: 1rem;
  }
  .projectName {
    font-size: 1.2rem;
  }
  .projectTechnologies {
    font-size: 0.5rem;
  }
}
