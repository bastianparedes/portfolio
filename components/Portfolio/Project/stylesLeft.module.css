.imageContainer,
.image {
  grid-column: 6 / -1;
}

.projectName {
  grid-column: 1 / 6;
  text-align: left;
}

.projectDescriptionContainer {
  grid-column: 1 / 9;
}

.projectTechnologies {
  grid-column: 1 / 6;
  text-align: left;
}

.linksContainer {
  grid-column: 1 / 3;
}

@media screen and (max-width: 768px) {
  .linksContainer {
    grid-column: 1 / 3;
  }
}

@media screen and (max-width: 480px) {
  .linksContainer {
    grid-column: 1 / 5;
  }
}
