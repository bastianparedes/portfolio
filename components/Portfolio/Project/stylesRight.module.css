.imageContainer,
.image {
  grid-column: 1 / 8;
}

.projectName {
  grid-column: 8 / -1;
  text-align: right;
}

.projectDescriptionContainer {
  grid-column: 5 / -1;
}

.projectTechnologies {
  grid-column: 8 / -1;
  text-align: right;
}

.linksContainer {
  grid-column: 11 / -1;
  direction: rtl;
}
