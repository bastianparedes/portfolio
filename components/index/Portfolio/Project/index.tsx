import React from 'react';

import { classNames } from 'bastianparedes/utils';
import Image from 'next/legacy/image';
import { VscGithub, VscSignOut } from 'react-icons/vsc';

import styles from './styles.module.scss';
import stylesLeft from './stylesLeft.module.scss';
import stylesRight from './stylesRight.module.scss';

interface ProjectProps {
  index: number;
  name: string;
  link?: string;
  image?: string;
  github?: string;
  description: string;
  technologies: string[];
}

const Project = ({
  index,
  name,
  link,
  image,
  github,
  description,
  technologies
}: ProjectProps): JSX.Element => {
  const stylesSide = index % 2 === 0 ? stylesRight : stylesLeft;

  return (
    <div className={styles.projectCard}>
      {link !== undefined ? (
        <a
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
          href={link}
          rel="noreferrer noopener"
          target="_blank"
        >
          {image !== undefined && (
            <Image
              alt={image}
              layout="fill"
              objectFit="contain"
              src={'/images/projects/' + image}
            />
          )}
        </a>
      ) : (
        <div
          className={classNames(
            styles.imageContainer,
            stylesSide.imageContainer
          )}
        >
          {image !== undefined && (
            <Image
              alt={image}
              layout="fill"
              objectFit="contain"
              src={'/images/projects/' + image}
            />
          )}
        </div>
      )}

      <h2 className={classNames(styles.projectName, stylesSide.projectName)}>
        {name}
      </h2>
      <div
        className={classNames(
          styles.projectDescriptionContainer,
          stylesSide.projectDescriptionContainer
        )}
      >
        <p className={styles.projectDescription}>{description}</p>
      </div>
      <span
        className={classNames(
          styles.projectTechnologies,
          stylesSide.projectTechnologies
        )}
      >
        {technologies.join(', ')}
      </span>
      <div
        className={classNames(styles.linksContainer, stylesSide.linksContainer)}
      >
        {link !== undefined && (
          <a
            className={styles.linkWebsite}
            href={link}
            rel="noreferrer noopener"
            target="_blank"
          >
            <VscSignOut />
          </a>
        )}
        {github !== undefined && (
          <a
            className={styles.linkGithub}
            href={github}
            rel="noreferrer noopener"
            target="_blank"
          >
            <VscGithub />
          </a>
        )}
      </div>
    </div>
  );
};

export default Project;
