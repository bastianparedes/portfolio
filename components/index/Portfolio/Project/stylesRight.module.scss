.imageContainer,
.image {
  grid-column: 1 / 8;
}

.projectName {
  grid-column: 8 / -1;
  text-align: right;
}

.projectDescriptionContainer {
  grid-column: 5 / -1;
}

.projectTechnologies {
  grid-column: 8 / -1;
  text-align: right;
}

.linksContainer {
  grid-column: 11 / -1;
  direction: rtl;
}

@media screen and (max-width: 768px) {
  .linksContainer {
    grid-column: 11 / -1;
  }
}

@media screen and (max-width: 480px) {
  .linksContainer {
    grid-column: 9 / -1;
  }
}
