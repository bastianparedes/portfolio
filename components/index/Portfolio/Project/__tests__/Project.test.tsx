import React from 'react';

import { render } from '@testing-library/react';

import Project from '..';

describe('<Project />', () => {
  const props = {
    description:
      'A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.',
    github: 'https://github.com/BastianParedes/falabella',
    image: 'falabella.png',
    index: 0,
    link: 'https://www.falabella.com/falabella-cl/',
    name: 'Falabella',
    technologies: [
      'HTML',
      'CSS',
      'JavaScript',
      'React.js',
      'Next.js',
      'TypeScript',
      'Dinamic Yield',
      'Adobe Experience Data Collection'
    ]
  };

  it('should render right', () => {
    const { container } = render(<Project {...props} index={0} />);
    expect(container).toMatchSnapshot();
  });

  it('should render left', () => {
    const { container } = render(<Project {...props} index={1} />);
    expect(container).toMatchSnapshot();
  });

  it('should render without link', () => {
    const { container } = render(<Project {...props} link="" />);
    expect(container).toMatchSnapshot();
  });

  it('should render without github', () => {
    const { container } = render(<Project {...props} github="" />);
    expect(container).toMatchSnapshot();
  });

  it('should render without link nor github', () => {
    const { container } = render(<Project {...props} link="" github="" />);
    expect(container).toMatchSnapshot();
  });
});
