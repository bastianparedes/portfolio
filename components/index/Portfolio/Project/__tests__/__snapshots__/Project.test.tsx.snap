// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Project /> should render left 1`] = `
<div>
  <div
    class="projectCard"
  >
    <a
      class="imageContainer imageContainer"
      href="https://www.falabella.com/falabella-cl/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="falabella.png"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </a>
    <h2
      class="projectName projectName"
    >
      Falabella
    </h2>
    <div
      class="projectDescriptionContainer projectDescriptionContainer"
    >
      <p
        class="projectDescription"
      >
        A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.
      </p>
    </div>
    <span
      class="projectTechnologies projectTechnologies"
    >
      HTML, CSS, JavaScript, React.js, Next.js, TypeScript, Dinamic Yield, Adobe Experience Data Collection
    </span>
    <div
      class="linksContainer linksContainer"
    >
      <a
        class="linkWebsite"
        href="https://www.falabella.com/falabella-cl/"
        rel="noreferrer noopener"
        target="_blank"
      />
      <a
        class="linkGithub"
        href="https://github.com/BastianParedes/falabella"
        rel="noreferrer noopener"
        target="_blank"
      />
    </div>
  </div>
</div>
`;

exports[`<Project /> should render right 1`] = `
<div>
  <div
    class="projectCard"
  >
    <a
      class="imageContainer imageContainer"
      href="https://www.falabella.com/falabella-cl/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="falabella.png"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </a>
    <h2
      class="projectName projectName"
    >
      Falabella
    </h2>
    <div
      class="projectDescriptionContainer projectDescriptionContainer"
    >
      <p
        class="projectDescription"
      >
        A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.
      </p>
    </div>
    <span
      class="projectTechnologies projectTechnologies"
    >
      HTML, CSS, JavaScript, React.js, Next.js, TypeScript, Dinamic Yield, Adobe Experience Data Collection
    </span>
    <div
      class="linksContainer linksContainer"
    >
      <a
        class="linkWebsite"
        href="https://www.falabella.com/falabella-cl/"
        rel="noreferrer noopener"
        target="_blank"
      />
      <a
        class="linkGithub"
        href="https://github.com/BastianParedes/falabella"
        rel="noreferrer noopener"
        target="_blank"
      />
    </div>
  </div>
</div>
`;

exports[`<Project /> should render without github 1`] = `
<div>
  <div
    class="projectCard"
  >
    <a
      class="imageContainer imageContainer"
      href="https://www.falabella.com/falabella-cl/"
      rel="noreferrer noopener"
      target="_blank"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="falabella.png"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </a>
    <h2
      class="projectName projectName"
    >
      Falabella
    </h2>
    <div
      class="projectDescriptionContainer projectDescriptionContainer"
    >
      <p
        class="projectDescription"
      >
        A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.
      </p>
    </div>
    <span
      class="projectTechnologies projectTechnologies"
    >
      HTML, CSS, JavaScript, React.js, Next.js, TypeScript, Dinamic Yield, Adobe Experience Data Collection
    </span>
    <div
      class="linksContainer linksContainer"
    >
      <a
        class="linkWebsite"
        href="https://www.falabella.com/falabella-cl/"
        rel="noreferrer noopener"
        target="_blank"
      />
    </div>
  </div>
</div>
`;

exports[`<Project /> should render without image 1`] = `
<div>
  <div
    class="projectCard"
  >
    <a
      class="imageContainer imageContainer"
      href="https://www.falabella.com/falabella-cl/"
      rel="noreferrer noopener"
      target="_blank"
    />
    <h2
      class="projectName projectName"
    >
      Falabella
    </h2>
    <div
      class="projectDescriptionContainer projectDescriptionContainer"
    >
      <p
        class="projectDescription"
      >
        A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.
      </p>
    </div>
    <span
      class="projectTechnologies projectTechnologies"
    >
      HTML, CSS, JavaScript, React.js, Next.js, TypeScript, Dinamic Yield, Adobe Experience Data Collection
    </span>
    <div
      class="linksContainer linksContainer"
    >
      <a
        class="linkWebsite"
        href="https://www.falabella.com/falabella-cl/"
        rel="noreferrer noopener"
        target="_blank"
      />
      <a
        class="linkGithub"
        href="https://github.com/BastianParedes/falabella"
        rel="noreferrer noopener"
        target="_blank"
      />
    </div>
  </div>
</div>
`;

exports[`<Project /> should render without link 1`] = `
<div>
  <div
    class="projectCard"
  >
    <div
      class="imageContainer imageContainer"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="falabella.png"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </div>
    <h2
      class="projectName projectName"
    >
      Falabella
    </h2>
    <div
      class="projectDescriptionContainer projectDescriptionContainer"
    >
      <p
        class="projectDescription"
      >
        A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.
      </p>
    </div>
    <span
      class="projectTechnologies projectTechnologies"
    >
      HTML, CSS, JavaScript, React.js, Next.js, TypeScript, Dinamic Yield, Adobe Experience Data Collection
    </span>
    <div
      class="linksContainer linksContainer"
    >
      <a
        class="linkGithub"
        href="https://github.com/BastianParedes/falabella"
        rel="noreferrer noopener"
        target="_blank"
      />
    </div>
  </div>
</div>
`;

exports[`<Project /> should render without link nor github 1`] = `
<div>
  <div
    class="projectCard"
  >
    <div
      class="imageContainer imageContainer"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="falabella.png"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </div>
    <h2
      class="projectName projectName"
    >
      Falabella
    </h2>
    <div
      class="projectDescriptionContainer projectDescriptionContainer"
    >
      <p
        class="projectDescription"
      >
        A/B Tester de la plataforma. Desarrollo nuevas versiones de la página y comparo la antigua versión con la nueva para recolectar datos y así desplegar la que obtenga mejores resultados.
      </p>
    </div>
    <span
      class="projectTechnologies projectTechnologies"
    >
      HTML, CSS, JavaScript, React.js, Next.js, TypeScript, Dinamic Yield, Adobe Experience Data Collection
    </span>
    <div
      class="linksContainer linksContainer"
    />
  </div>
</div>
`;
